File: core/src/main/java/net/weweave/tubewarder/domain/Attachment.java
package net.weweave.tubewarder.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;

@Entity
public class Attachment extends AbstractPersistentObject {
    @ManyToOne
    private SendQueueItem sendQueueItem;
    private String filename;
    private String contentType;
    @Column(length = 100000)
    private String payload;

    public SendQueueItem getSendQueueItem() {
        return sendQueueItem;
    }

    public void setSendQueueItem(SendQueueItem sendQueueItem) {
        this.sendQueueItem = sendQueueItem;
    }

    public String getFilename() {
        return filename;
    }

    public void setFilename(String filename) {
        this.filename = filename;
    }

    public String getContentType() {
        return contentType;
    }

    public void setContentType(String contentType) {
        this.contentType = contentType;
    }

    public String getPayload() {
        return payload;
    }

    public void setPayload(String payload) {
        this.payload = payload;
    }
}

File: clients/php/src/Attachment.php
<?php
namespace Tubewarder;

class Attachment {
    private $filename;
    private $contentType;
    private $payload;

    public function __construct($filename = '', $payload = '', $contentType = '') {
        $this->filename = $filename;
        $this->payload = $payload;
        $this->contentType = $contentType;
    }

    /**
     * @return string
     */
    public function getFilename() {
        return $this->filename;
    }

    /**
     * @param string $filename
     */
    public function setFilename($filename) {
        $this->filename = $filename;
    }

    /**
     * @return string
     */
    public function getContentType() {
        return $this->contentType;
    }

    /**
     * @param string $contentType
     */
    public function setContentType($contentType) {
        $this->contentType = $contentType;
    }

    /**
     * @return string
     */
    public function getPayload() {
        return $this->payload;
    }

    /**
     * @param string $payload
     */
    public function setPayload($payload) {
        $this->payload = $payload;
    }

    /**
     * @param string $filename
     */
    public function setPayloadFromFile($filename) {
        $this->payload = base64_encode(file_get_contents($filename));
    }

    /**
     * @return array
     */
    public function getObject() {
        $o = array(
            'filename' => $this->getFilename(),
            'payload' => $this->getPayload(),
            'contentType' => $this->getContentType()
        );
        return $o;
    }
}